p_sets/3/generate.rb

```diff
@@ -0,0 +1,26 @@
+class RandomArrayGenerator
+  EXPONENT = 10**3
+  attr_reader :length, :seed
+
+  def initialize(length, seed = nil)
+    @length = length
+    @seed = seed
+  end
+
+  def array
+    Array.new(length) { (rand * EXPONENT).ceil }
+  end
+
+  def seed_rand
+    srand seed
```

Comment:
I'm being pedantic here, but I probably wouldn't rely on the calling code to call this function. I'd instead in `initialize` do something like `srand(seed) if seed` - then you don't need the `seed` instance variable either.

I don't have a great explanation of why I feel that - I guess it feels like part of the internal logic of the class which any code using it would have to do in the same order, so you might as well force that order to be followed.